packages/nimble-components/src/wafer-map/index.ts

```diff
@@ -134,53 +140,69 @@ export class WaferMap extends FoundationElement {
         values: []
     };
 
-    private eventCoordinator?: EventCoordinator;
-    private resizeObserver?: ResizeObserver;
+    private readonly eventCoordinator = new EventCoordinator(this);
+    private readonly resizeObserver = this.createResizeObserver();
 
     public override connectedCallback(): void {
         super.connectedCallback();
+        this.canvas.width = this.clientWidth;
+        this.canvas.height = this.clientHeight;
         this.canvasContext = this.canvas.getContext('2d', {
             willReadFrequently: true
         })!;
-        this.resizeObserver = this.createResizeObserver();
+        this.resizeObserver.observe(this);
+        this.waferMapUpdateTracker.trackAll();
     }
 
     public override disconnectedCallback(): void {
         super.disconnectedCallback();
-        this.resizeObserver!.unobserve(this);
+        this.resizeObserver.unobserve(this);
     }
 
     /**
      * @internal
      */
-    public render(): void {
-        this.renderQueued = false;
-        this.initializeInternalModules();
-        this.renderer?.drawWafer();
-    }
-
-    private queueRender(): void {
-        if (!this.$fastController.isConnected) {
-            return;
-        }
-        if (!this.renderQueued) {
-            this.renderQueued = true;
-            DOM.queueUpdate(() => this.render());
-        }
+    public setHoverData(
+        hoverWidth: number,
+        hoverHeight: number,
+        hoverOpacity: HoverDieOpacity,
+        hoverTransform: string
+    ): void {
+        this.hoverWidth = hoverWidth;
+        this.hoverHeight = hoverHeight;
+        this.hoverOpacity = hoverOpacity;
+        this.hoverTransform = hoverTransform;
     }
```

Comment:
It's not clear to me what the benefit of this method is. Other spots that mutate the wafermap state manipulate the properties directly like zoom-handler and hover-handler.

Since this method is only used from one spot and does not follow other patterns for manipulating wafermap state I would recommend we don't add it and the rendering module just sets the properties like before